lib/tesla/adapter/shared.ex

```diff
@@ -12,4 +12,13 @@ defmodule Tesla.Adapter.Shared do
 
   defp parse_chunk({:suspended, item, fun}), do: {:ok, item, fun}
   defp parse_chunk(_), do: :eof
+
+  @spec prepare_path(String.t() | nil, String.t() | nil) :: String.t()
+  def prepare_path(nil, nil), do: "/"
+  def prepare_path(nil, query), do: "/?" <> query
+  def prepare_path(path, nil), do: path
+  def prepare_path(path, query), do: path <> "?" <> query
+
+  @spec format_method(atom()) :: String.t()
+  def format_method(method), do: to_string(method) |> String.upcase()
```

Comment:
Are these functions used in any other adapter? Uppercase binary method seems gun-specific for example.

Reply:
Mint adapter will be upstreamed soon. These functions are used there too.

Reply:
👌